pkg/plugins/common/kustomize/v1/scaffolds/init.go

```diff
@@ -62,23 +62,44 @@ func (s *initScaffolder) Scaffold() error {
 		machinery.WithConfig(s.config),
 	)
 
-	return scaffold.Execute(
-		&rbac.Kustomization{},
-		&rbac.AuthProxyRole{},
-		&rbac.AuthProxyRoleBinding{},
-		&rbac.AuthProxyService{},
-		&rbac.AuthProxyClientRole{},
-		&rbac.RoleBinding{},
-		&rbac.LeaderElectionRole{},
-		&rbac.LeaderElectionRoleBinding{},
-		&rbac.ServiceAccount{},
-		&manager.Kustomization{},
-		&manager.Config{Image: imageName},
-		&manager.ControllerManagerConfig{},
-		&kdefault.Kustomization{},
-		&kdefault.ManagerAuthProxyPatch{},
-		&kdefault.ManagerConfigPatch{},
-		&prometheus.Kustomization{},
-		&prometheus.Monitor{},
-	)
+	if s.config.IsComponentConfig() {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&manager.ControllerManagerConfig{},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
+			&prometheus.Monitor{},
+		)
+	} else {
+		return scaffold.Execute(
+			&rbac.Kustomization{},
+			&rbac.AuthProxyRole{},
+			&rbac.AuthProxyRoleBinding{},
+			&rbac.AuthProxyService{},
+			&rbac.AuthProxyClientRole{},
+			&rbac.RoleBinding{},
+			&rbac.LeaderElectionRole{},
+			&rbac.LeaderElectionRoleBinding{},
+			&rbac.ServiceAccount{},
+			&manager.Kustomization{},
+			&manager.Config{Image: imageName},
+			&kdefault.Kustomization{},
+			&kdefault.ManagerAuthProxyPatch{},
+			&kdefault.ManagerConfigPatch{},
+			&prometheus.Kustomization{},
```

Comment:
```suggestion
			&kdefault.ManagerAuthProxyPatch{},
			&prometheus.Kustomization{},
```

Reply:
As per the discussion, I will be making the patch conditional based. If the user specifies, the `--component-config` flag then only `kustomization.yml` file will have that patch. I am working on changes.

Reply:
@laxmikantbpandhare 

The `&kdefault.ManagerConfigPatch{},` is used in the default scaffold?
I mean, have we any reason to do the scaffold if/when we are not looking the use of the ComponentConfig?

If not, then we do not need to scaffold so, we need to check it.

Reply:
@camilamacedo86 @SayakMukhopadhyay - the scaffold of `kustomization.yaml` from `/config/default` location scaffolds some of the part depending upon `componenet-config`. We are planning to use the same login in `/config/manager` too. I will update the PR, and every question will be answered.

Reply:
Code changes completed. Working on docs and testing part.

Reply:
There is no need to change in `/config/default`. It is because the `kustomization.yml` file commented out the file `manager_config_patch.yaml` whenever there is no flag passed. So, this file only gets triggered if the flag is passed through the command line.

Reply:
Yeah, its true that the `kustomization.yaml` has the patch line commented out as I noted in this [comment](https://github.com/kubernetes-sigs/kubebuilder/pull/2826#pullrequestreview-1046430684). The reason I would prefer this file to be conditionally scaffolded too as the presence of this file could confuse first time users as the file was not _strictly_ getting used anywhere.

Yeah, its mentioned in the commented line in `kustomization.yaml` but so is `manager_webhook_patch.yaml` and `webhookcainjection_patch.yaml` but those files are not created unless needed.

Reply:
Agree with you, if the `--component-config` field is not passed then these two files are getting created unnecessarily.
Let's see what @camilamacedo86 says on this.

Reply:
I think it makes sense close this one as sorted out.